### ISSUE TITLE
WLED does not respond to stream from PC

### ISSUE THREAD

**Issue by thejustsoul** [Victoare/SR-WLED-audio-server-win#5]
In WLED 0.15.0, Audio Reactive Usermod support has been officially added. I'm trying to set up data reception from a PC, but WLED doesn't respond.

At the same time, transmission from another WLED works fine (there is an analog microphone), but neither the official nor the MoonModules version of WLED respond to the stream from the PC. With LedFX, everything is transmitted normally.

What am I doing wrong? How can I make SR-WLED-audio-server work with WLED 0.15.0+?

![изображение](https://github.com/user-attachments/assets/c42d9328-fa94-44a2-b593-515ecec521ff)
![изображение](https://github.com/user-attachments/assets/917b21b3-4be1-4420-ab0b-5136eb444126)![изображение](https://github.com/user-attachments/assets/f1998a23-b09a-48eb-b097-7d3a80206f8c)
![изображение](https://github.com/user-attachments/assets/46a50fbf-29e3-4892-b242-669f46fa2f97)

**Comment by Victoare** [comment]
If you set the "local IP" as your PC ip address then it should work without disabling the other interfaces.

I will test the 0.15.0 version soon.

There is a possibility that the official and the moonmodules versions uses different ranges (and therefore that overload effect)
If you have two modules at hand you can check the two versions side-by side. They should receive the same data at the same time and produce the same effects (I will also do this soon)

One question though: what hardware do you use? esp8266 or esp32?

**Comment by Victoare** [comment]
Yes, I saw the ip, but some folks get confused and set the wled modules address there.
Checeked today with the official wled 0.15.0 (esp8266 160mhz) version and it is working as intended.
There is an automatic gain control implemented in the code to "max-out" the freq movement so there will be a frequency bar that will go almost to the top regardles of the volume of the media. Not sure if this is what your issue is.
Maybe an option to turn AGC off and let the user play with attenuation and compression settings or give more control over the AGC itself, but not really planning on that yet.

**Comment by thejustsoul** [comment]
I took a new board on EPS8266, connected it to another WS2812B strip, flashed it with WLED 0.15.0 160 Mhz, turned on Audio Reactive and Recieve mode. Reset the network, connected via WiFi (instead of cable), changed the IP in the program to the adapter address. Nothing changed. Can I see how it works for you? For example, on the Gravcenter, Gravmetric, Gravimeter effects. For me, they just light up in one color and rarely blink.

**Comment by Victoare** [comment]
The main difference that LedFX drives the individual leds directly while the soundReactive usermod receives information about the frequency structure and the effects decides what to do with it. 

I am aware that a couple effect is not or hardly working. Sadly there is no proper documentation and I mostly gathered the current info reading wled's code and try to understand it.

For led-strip I usually use it with DJ Light with "cady factory" settings:
![image](https://github.com/user-attachments/assets/c87eb480-36b5-49c2-abaf-585295e84605)

You can check the delay and sensitivity to the music here: https://youtu.be/b4_3JbWpXxM
PingPong ball led strip using DJLight, the left 2D matrix using GEQ the right one using "funky plank", the infinity mirror a fourth effect (not sure which). Each one using its own esp8266 module.
Btw this is an older video and DjLight is much more colorful with the current version.
The delay should be minimal, like less than 50ms. If you experiencing longer delays then there will be some other (possibly network related) problem.

There is no strict answer for the "correct parameter". I usually disable all limiter or AGC on the wled side and on the app side I am using log scale (without it, only the first bar will react the bass and mid frequencies and the others will react to higher freqs) and "square root (energy)" as value scaling.
